src/pudl/analysis/ml_tools.py

```diff
@@ -0,0 +1,181 @@
+"""This module provides shared tooling that can be used by all record linkage models."""
```

Comment:
Does this tooling still only generalize to record linkage models, or can it just read "all PUDL models" or something?